openhcl/underhill_core/src/nvme_manager.rs

```diff
@@ -115,16 +131,38 @@ impl NvmeManager {
         // if nvme_keepalive == true { return }
         self.client.sender.send(Request::Shutdown {
             span: tracing::info_span!("shutdown_nvme_manager"),
-            nvme_keepalive,
+            nvme_keepalive: self.nvme_keepalive,
         });
         self.task.await;
     }
+
+    /// Save NVMe manager's state during servicing.
+    pub async fn save(&self) -> anyhow::Result<NvmeManagerSavedState> {
+        // NVMe manager has no own data to save, everything will be done
+        // in the Worker task which can be contacted through Client.
+        if self.nvme_keepalive {
+            self.client().save().await
+        } else {
+            // If nvme_keepalive was explicitly disabled,
+            // return an error which is non-fatal indication
+            // that there is no save data.
+            Err(anyhow::Error::from(SaveRestoreError::ExplicitlyDisabled {}))
```

Comment:
Not blocking for this PR:
Is this error logged anywhere? If so, please consider logging it as a warning.

Reply:
It is not completely untraceable, the nvme_keepalive flag will be printed in shutdown_nvme_vfio span, but I agree - this case deserves a dedicated trace.